src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -205,6 +206,17 @@ private class TaskLauncherActor(
   }
 
   private[this] def receiveTaskLaunchNotification: Receive = {
+    case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
+      // Reschedule instance with provision timeout..
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+
+          // We don't await the future. It will trigger an instance update event.
+          logger.info(s"Reschedule ${instance.instanceId} because of provision timeout.")
+          instanceTracker.forceExpunge(instance.instanceId)
```

Comment:
don't we need to do something different for resident tasks?

Reply:
I'm not sure how we can reschedule task that do have a reservation but do not become running in time. I would just expunge the whole thing.

This should just be a short term solution. I really don't want this to be a blocker for phase 2.

Reply:
@meichstedt, what is your take? We won't have a proper solution without incarnation counts and reconciliations as described in [MARATHON-8235](https://jira.mesosphere.com/browse/MARATHON-8235).

Reply:
Shoot. I just realized that this might delete reservations on restarts as well. I'm inclined to drop the timeout for resident apps and pods.

Reply:
why don't we just decommission/stop? I think we should not use forceExpunge... ideally anywhere. It feel super hacky to me...

Reply:
Ideally we would but we don't kill tasks based on the goal state yet. That's phase 3.

Reply:
oh that's right. And wouldn't the kill happen anyway just because we overscale? (potentially)

I think that in the old implementation (before phase 2) we did not kill as well. We just eventually overscaled...

Reply:
Well the problem is this: if a resident task fails, the instance is kept and used to reschedule that task. If we expunge the instance in the case of this timeout, we would destroy persistent data, as long as the reservation/volume metadata is stored within the instance.

I'd not be OK with expunging even if it's just temporarily – it breaks resident tasks.

My suggestion is to handle this timeout with – tadaa – a surrogate signal. So instead of expunging the instance via `instanceTracker.forceExpunge(instance.instanceId)`, I'd build a GONE status like @timcharper did here https://github.com/mesosphere/marathon/pull/6290/files#diff-5e3e6fe795e2ed487104e6855e38ea32L69

add proper logging on level WARN that we are faking a Mesos task status here, and do
`instanceTracker.updateStatus(instance, goneStatus, now)`

That way, Marathon will keep the instance, and re-schedule as if that task was simply reported as GONE (meaning it'll increment the attempt count on the task etc). That's closest to the eager reconciliation described in MARATHON-8235 as we can get now.

Reply:
I did a prototype in an older commit https://github.com/mesosphere/marathon/pull/6418/commits/e2c07629c007c95191193c98a577b42d180c1293. I'm not that convinced yet. For ephemeral task the expunge has the same effect as we have one master.

I'll have to dig into the current implementation for resident tasks.

Reply:
Ok, it seems we just [overscale on master](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala#L208). On master, if the timeout hits we simple schedule a new launch. If the old one does succeed we overscaled.